contracts/src/registry/L1ETHRegistry.sol

```diff
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.13;
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+
+import {ERC1155Singleton} from "./ERC1155Singleton.sol";
+import {IERC1155Singleton} from "./IERC1155Singleton.sol";
+import {IRegistry} from "./IRegistry.sol";
+import {IRegistryDatastore} from "./IRegistryDatastore.sol";
+import {BaseRegistry} from "./BaseRegistry.sol";
+import {PermissionedRegistry} from "./PermissionedRegistry.sol";
+
+/**
+ * @title L1ETHRegistry
+ * @dev L1 contract for .eth that holds ejected .eth names only.
+ * Unlike the L2 ETHRegistry, this registry does not handle new registrations directly,
+ * but receives names that have been ejected from L2.
+ */
+contract L1ETHRegistry is PermissionedRegistry, AccessControl {
+    bytes32 public constant EJECTION_CONTROLLER_ROLE = keccak256("EJECTION_CONTROLLER_ROLE");
+    bytes32 public constant RENEWAL_CONTROLLER_ROLE = keccak256("RENEWAL_CONTROLLER_ROLE");
+
+    error NameExpired(uint256 tokenId);
+    error CannotReduceExpiration(uint64 oldExpiration, uint64 newExpiration);
+
+    event NameEjected(uint256 indexed tokenId, address owner, uint64 expires);
+    event NameRenewed(uint256 indexed tokenId, uint64 newExpiration, address renewedBy);
+    event NameRelinquished(uint256 indexed tokenId, address relinquishedBy);
+    event NameMigratedToL2(uint256 indexed tokenId, address sendTo);
+    event FallbackResolverSet(address resolver);
```

Comment:
I don't think we need an event for this, since the registry will emit an event in any case.